zkvm/src/txlog.rs

```diff
@@ -66,43 +47,15 @@ impl UTXO {
 impl TxID {
     /// Computes TxID from a tx log
     pub fn from_log(list: &[Entry]) -> Self {
-        let t = Transcript::new(b"ZkVM.txid");
```

Comment:
this was computing the hash almost in-place, w/o allocating all boxed nodes in memory. I think we should have a separate entry point to MerkleTree to retain this lightweight computation. Something like `MerkleTree::root(list)` in addition to `MerkleTree::build(list)` ("build" probably better reflects the "heaviness" of the structure than a typical `::new`).

Reply:
obviously, `MerkleTree::root()` is a redundant version of `MerkleTree::build().root()`, but it's doing no allocations and then can be even rewritten to be iterative instead of recursive.